src/app/components/book-list/book-list.component.ts

```diff
@@ -42,16 +51,21 @@ export class BookListComponent implements OnInit {
 
         dialogRef.componentInstance.action = FormActions.Add;
 
-        dialogRef.afterClosed().subscribe(result => {
-            if(result) {
-                this.addBook(result); 
-            }  
-        })
+        this.subscriptions.push(
+            dialogRef.afterClosed().subscribe(result => {
+                if(result) {
```

Comment:
Code style.

Reply:
What is actually wrong with code style here?

Reply:
Space between if and (